Print the Bitcoin address to the terminal as a QR code (#547)
comit-network/xmr-btc-swap: CHANGELOG.md

## [Unreleased]

### Added

- Printing the deposit address to the terminal as a QR code.

>   Users who are doing automated scripts or integrating with other software can use the `--json` flag to avoid this behaviour. It won't print a QR code if the `--json` flag is being passed to the application.

Reviewer: From the way you worded this, I think there was a bit of a misunderstanding here. The `--json` flag doesn't exist to avoid the QRCode printing. The `--json` flag toggles whether or not the logs are printed in JSON format. It will make more sense once you try it!

Hence, what we should mention in the changelog is the fact that `--json` is conceptually incompatible with printing a QRCode and hence will be skipped.

Something like:

```suggestion
  To not break automated scripts or integrations with other software, this behaviour is disabled if `--json` is passed to the application.
```

Author: Yeah I tried the `--json` flag to test it, its prints the logs in JSON format. This was a bad wording choice by me.

- Fixed an issue where Alice would not verify if Bob's Bitcoin lock transaction is semantically correct, i.e. pays the agreed upon amount to an output owned by both of them.
  Fixing this required a **breaking change** on the network layer and hence old versions are not compatible with this version.

> [unreleased]: https://github.com/comit-network/xmr-btc-swap/compare/0.7.0...HEAD

Reviewer: Why was this updated?

Author: Ughh I've been a victimized by Prettier this time :smile: (https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)

Reviewer: We use [`dprint`](https://dprint.dev/) to format things in this repo. If you've ever been annoyed by how slow prettier is, give it a go :)
